packages/matter.js/src/environment/VariableService.ts

```diff
@@ -243,8 +244,8 @@ export function merge(a: Record<string, any>, b: Record<string, any>) {
     for (const key in b) {
         const aval = a[key];
         const bval = b[key];
-        if (typeof bval === "object") {
```

Comment:
This is an example of where using "any" instead of "unknown" causes trouble, I'd probably fix that too.